h/api/schemas.py

```diff
@@ -367,10 +362,72 @@ def validate(self, data):
         return appstruct
 
 
+def _document(document, claimant):
+    """
+    Return document meta and document URI data from the given document dict.
+
+    Transforms the "document" dict that the client posts into a convenient
+    format for creating DocumentURI and DocumentMeta objects later.
+
+    """
+    document = document or {}
+    document_uri_dicts = parse_document_claims.document_uris_from_data(
+        copy.deepcopy(document),
+        claimant=claimant)
+    document_meta_dicts = parse_document_claims.document_metas_from_data(
+        copy.deepcopy(document),
+        claimant=claimant)
+    return {
+        'document_uri_dicts': document_uri_dicts,
+        'document_meta_dicts': document_meta_dicts
+    }
+
+
 def _format_jsonschema_error(error):
     """Format a :py:class:`jsonschema.ValidationError` as a string."""
     if error.path:
         dotted_path = '.'.join([str(c) for c in error.path])
         return '{path}: {message}'.format(path=dotted_path,
                                           message=error.message)
     return error.message
+
+
+def _remove_protected_fields(appstruct):
+    # Some fields are not to be set by the user, ignore them.
+    for field in ['created', 'updated', 'user', 'id', 'links']:
+        appstruct.pop(field, None)
+
+
+def _shared(permissions, userid):
+    """
+    Return True if the given permissions object represents shared permissions.
+
+    Return False otherwise.
+
+    Reduces the client's complex permissions dict to a simple shared boolean.
+
+    :param permissions: the permissions dict sent by the client in an
+        annotation create or update request
+    :type permissions: dict
+
+    :param userid: the userid of the user who created the annotation
+    :type userid: unicode
+
+    """
+    return permissions['read'] != [userid]
```

Comment:
Is there a reason this is expressed this way round rather than `permissions['read'] == ['group:__world__']`? Defaulting to closed in the event of unrecognised data seems a bit safer than defaulting to open.